core/src/test/java/org/elasticsearch/index/engine/ShadowEngineTests.java

```diff
@@ -217,24 +215,31 @@ protected InternalEngine createInternalEngine(IndexSettings indexSettings, Store
     }
 
     protected InternalEngine createInternalEngine(IndexSettings indexSettings, Store store, Path translogPath, MergePolicy mergePolicy) {
-        return new InternalEngine(config(indexSettings, store, translogPath, mergePolicy), true);
+        EngineConfig config = config(indexSettings, store, translogPath, mergePolicy);
+        return new InternalEngine(config);
     }
 
     public EngineConfig config(IndexSettings indexSettings, Store store, Path translogPath, MergePolicy mergePolicy) {
         IndexWriterConfig iwc = newIndexWriterConfig();
+        final EngineConfig.OpenMode openMode;
+        try {
+            if (Lucene.indexExists(store.directory()) == false) {
+                openMode = EngineConfig.OpenMode.CREATE_INDEX_AND_TRANSLOG;
```

Comment:
I don't get what you mean either way if use `CREATE_INDEX_AND_TRANSLOG` we have to add a commit in `InternalEngine#openTranslog` id we use `OPEN_INDEX_CREATE_TRANSLOG` we add a commit in the `InternalEngine#ctor` line 160 
```Java
if (openMode == EngineConfig.OpenMode.OPEN_INDEX_CREATE_TRANSLOG) {
  // make sure we point at the latest translog from now on..
   commitIndexWriter(writer, translog, lastCommittedSegmentInfos.getUserData().get(SYNC_COMMIT_ID));
}
```